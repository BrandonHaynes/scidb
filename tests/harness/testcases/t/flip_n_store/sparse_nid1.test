--setup
--start-query-logging
# Ticket number: 1624
# Tests on 3D-sparse NID arrays.
# Tests covering aggregates,joins,subarray,slice,thin on NID single/multi attribute 3D arrays

create array A <a1: double > [x=0:5,3,0]
create array NID <x:int64> [a1(double)=6,3,0, a2(double)=6,3,0 ,a3(double)=6,3,0]
store(apply(apply(build(A,13.5*6/x),a2,a1*5.72),a3,a2*0.5),A_derivative)


--test
--set-format sparse
store(redimension(A_derivative,NID),NID)
store(subarray(NID,16.2,92.664,46.332,81,463.32,231.66),AAA)
--error --code=scidb::SCIDB_SE_SYNTAX::SCIDB_LE_AMBIGUOUS_DIMENSION "cross_join(NID,AAA,a1,a1)"

slice(NID,a1,81)
slice(NID,a2,154.44)
join(slice(NID,a2,154.44),slice(NID,a1,81))
cross_join(join(slice(NID,a2,154.44),slice(NID,a1,81)),NID)
--error --code=scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_ARRAYS_NOT_CONFORMANT "cross_join(AAA,cast(AAA,<x:int64> [a11(double)=5,3,0,a22(double)=4,3,0,a33(double)=4,3,0]),a1,a22)"
--error --code=scidb::SCIDB_SE_TYPE::SCIDB_LE_CANT_FIND_IMPLICIT_CONVERTER "thin(NID,16.2,3,92.664,3,46.332,3)"

store(apply(apply(NID,y,x*100.5),z,y*12),NID_multi)
aggregate(NID_multi,sum(x))
aggregate(NID_multi,sum(y))
aggregate(NID_multi,sum(z))
aggregate(NID_multi,sum(x))

--aql select cos(x),sin(2*y) from NID_multi
--aql select cos(x),sin(2*y),sqrt(z) from NID_multi
--aql select sum(x+2), avg(x+1) from NID_multi
--aql select sum(x), avg(x) from NID_multi
--aql select cos(x),sin(2*y) from NID_multi

aggregate(NID_multi,sum(z),var(x),min(x),max(z))
join(NID,NID_multi)
cross_join(NID,NID_multi)


--cleanup
remove(A)
remove(NID)
remove(A_derivative)
remove(AAA)
remove(NID_multi)

--stop-query-logging
