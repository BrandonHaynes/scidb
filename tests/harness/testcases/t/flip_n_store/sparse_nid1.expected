SCIDB QUERY : <create array A <a1: double > [x=0:5,3,0]>
Query was executed successfully

SCIDB QUERY : <create array NID <x:int64> [a1(double)=6,3,0, a2(double)=6,3,0 ,a3(double)=6,3,0]>
Query was executed successfully

SCIDB QUERY : <store(apply(apply(build(A,13.5*6/x),a2,a1*5.72),a3,a2*0.5),A_derivative)>
{x} a1,a2,a3
{0} inf,inf,inf
{1} 81,463.32,231.66
{2} 40.5,231.66,115.83
{3} 27,154.44,77.22
{4} 20.25,115.83,57.915
{5} 16.2,92.664,46.332

SCIDB QUERY : <store(redimension(A_derivative,NID),NID)>
{0,0,0}[[[{0,0,0}(5),{1,1,1}(4),{2,2,2}(3)]]];{3,3,3}[[[{3,3,3}(2),{4,4,4}(1),{5,5,5}(0)]]]

SCIDB QUERY : <store(subarray(NID,16.2,92.664,46.332,81,463.32,231.66),AAA)>
[[[{1,0,0}(4),{2,1,1}(3)]]];
[[[{3,2,2}(2)]]];{4,3,3}[[[{4,3,3}(1)]]]

SCIDB QUERY : <cross_join(NID,AAA,a1,a1)>
[An error expected at this place for the query "cross_join(NID,AAA,a1,a1)". And it failed with error code = scidb::SCIDB_SE_SYNTAX::SCIDB_LE_AMBIGUOUS_DIMENSION. Expected error code = scidb::SCIDB_SE_SYNTAX::SCIDB_LE_AMBIGUOUS_DIMENSION.]

SCIDB QUERY : <slice(NID,a1,81)>
{4,4}[[{4,4}(1)]]

SCIDB QUERY : <slice(NID,a2,154.44)>
{2,2}[[{2,2}(3)]]

SCIDB QUERY : <join(slice(NID,a2,154.44),slice(NID,a1,81))>
[[]]

SCIDB QUERY : <cross_join(join(slice(NID,a2,154.44),slice(NID,a1,81)),NID)>
[[[[[]]]]]

SCIDB QUERY : <cross_join(AAA,cast(AAA,<x:int64> [a11(double)=5,3,0,a22(double)=4,3,0,a33(double)=4,3,0]),a1,a22)>
[An error expected at this place for the query "cross_join(AAA,cast(AAA,<x:int64> [a11(double)=5,3,0,a22(double)=4,3,0,a33(double)=4,3,0]),a1,a22)". And it failed with error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_ARRAYS_NOT_CONFORMANT. Expected error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_ARRAYS_NOT_CONFORMANT.]

SCIDB QUERY : <thin(NID,16.2,3,92.664,3,46.332,3)>
[An error expected at this place for the query "thin(NID,16.2,3,92.664,3,46.332,3)". And it failed with error code = scidb::SCIDB_SE_TYPE::SCIDB_LE_CANT_FIND_IMPLICIT_CONVERTER. Expected error code = scidb::SCIDB_SE_TYPE::SCIDB_LE_CANT_FIND_IMPLICIT_CONVERTER.]

SCIDB QUERY : <store(apply(apply(NID,y,x*100.5),z,y*12),NID_multi)>
{0,0,0}[[[{0,0,0}(5,502.5,6030),{1,1,1}(4,402,4824),{2,2,2}(3,301.5,3618)]]];{3,3,3}[[[{3,3,3}(2,201,2412),{4,4,4}(1,100.5,1206),{5,5,5}(0,0,0)]]]

SCIDB QUERY : <aggregate(NID_multi,sum(x))>
{0}[{0}(15)]

SCIDB QUERY : <aggregate(NID_multi,sum(y))>
{0}[{0}(1507.5)]

SCIDB QUERY : <aggregate(NID_multi,sum(z))>
{0}[{0}(18090)]

SCIDB QUERY : <sum(NID_multi)>
{0}[{0}(15)]

SCIDB QUERY : <select cos(x),sin(2*y) from NID_multi>
{0,0,0}[[[{0,0,0}(0.283662,-0.304724),{1,1,1}(-0.653644,-0.245194),{2,2,2}(-0.989992,-0.184722)]]];{3,3,3}[[[{3,3,3}(-0.416147,-0.123543),{4,4,4}(0.540302,-0.0618903),{5,5,5}(1,0)]]]

SCIDB QUERY : <select cos(x),sin(2*y),sqrt(z) from NID_multi>
{0,0,0}[[[{0,0,0}(0.283662,-0.304724,77.6531),{1,1,1}(-0.653644,-0.245194,69.455),{2,2,2}(-0.989992,-0.184722,60.1498)]]];{3,3,3}[[[{3,3,3}(-0.416147,-0.123543,49.1121),{4,4,4}(0.540302,-0.0618903,34.7275),{5,5,5}(1,0,0)]]]

SCIDB QUERY : <select sum(x+2), avg(x+1) from NID_multi>
{0}[{0}(27,3.5)]

SCIDB QUERY : <select sum(x), avg(x) from NID_multi>
{0}[{0}(15,2.5)]

SCIDB QUERY : <select cos(x),sin(2*y) from NID_multi>
{0,0,0}[[[{0,0,0}(0.283662,-0.304724),{1,1,1}(-0.653644,-0.245194),{2,2,2}(-0.989992,-0.184722)]]];{3,3,3}[[[{3,3,3}(-0.416147,-0.123543),{4,4,4}(0.540302,-0.0618903),{5,5,5}(1,0)]]]

SCIDB QUERY : <aggregate(NID_multi,sum(z),var(x),min(x),max(z))>
{0}[{0}(18090,3.5,0,6030)]

SCIDB QUERY : <join(NID,NID_multi)>
{0,0,0}[[[{0,0,0}(5,5,502.5,6030),{1,1,1}(4,4,402,4824),{2,2,2}(3,3,301.5,3618)]]];{3,3,3}[[[{3,3,3}(2,2,201,2412),{4,4,4}(1,1,100.5,1206),{5,5,5}(0,0,0,0)]]]

SCIDB QUERY : <cross_join(NID,NID_multi)>
{0,0,0,0,0,0}[[[[[[{0,0,0,0,0,0}(5,5,502.5,6030),{0,0,0,1,1,1}(5,4,402,4824),{0,0,0,2,2,2}(5,3,301.5,3618)]]],[[[{1,1,1,0,0,0}(4,5,502.5,6030),{1,1,1,1,1,1}(4,4,402,4824),{1,1,1,2,2,2}(4,3,301.5,3618)]]],[[[{2,2,2,0,0,0}(3,5,502.5,6030),{2,2,2,1,1,1}(3,4,402,4824),{2,2,2,2,2,2}(3,3,301.5,3618)]]]]]];{0,0,0,3,3,3}[[[[[[{0,0,0,3,3,3}(5,2,201,2412),{0,0,0,4,4,4}(5,1,100.5,1206),{0,0,0,5,5,5}(5,0,0,0)]]],[[[{1,1,1,3,3,3}(4,2,201,2412),{1,1,1,4,4,4}(4,1,100.5,1206),{1,1,1,5,5,5}(4,0,0,0)]]],[[[{2,2,2,3,3,3}(3,2,201,2412),{2,2,2,4,4,4}(3,1,100.5,1206),{2,2,2,5,5,5}(3,0,0,0)]]]]]];{3,3,3,0,0,0}[[[[[[{3,3,3,0,0,0}(2,5,502.5,6030),{3,3,3,1,1,1}(2,4,402,4824),{3,3,3,2,2,2}(2,3,301.5,3618)]]],[[[{4,4,4,0,0,0}(1,5,502.5,6030),{4,4,4,1,1,1}(1,4,402,4824),{4,4,4,2,2,2}(1,3,301.5,3618)]]],[[[{5,5,5,0,0,0}(0,5,502.5,6030),{5,5,5,1,1,1}(0,4,402,4824),{5,5,5,2,2,2}(0,3,301.5,3618)]]]]]];{3,3,3,3,3,3}[[[[[[{3,3,3,3,3,3}(2,2,201,2412),{3,3,3,4,4,4}(2,1,100.5,1206),{3,3,3,5,5,5}(2,0,0,0)]]],[[[{4,4,4,3,3,3}(1,2,201,2412),{4,4,4,4,4,4}(1,1,100.5,1206),{4,4,4,5,5,5}(1,0,0,0)]]],[[[{5,5,5,3,3,3}(0,2,201,2412),{5,5,5,4,4,4}(0,1,100.5,1206),{5,5,5,5,5,5}(0,0,0,0)]]]]]]

SCIDB QUERY : <remove(A)>
Query was executed successfully

SCIDB QUERY : <remove(NID)>
Query was executed successfully

SCIDB QUERY : <remove(A_derivative)>
Query was executed successfully

SCIDB QUERY : <remove(AAA)>
Query was executed successfully

SCIDB QUERY : <remove(NID_multi)>
Query was executed successfully

